src/opers.c

```diff
@@ -1831,788 +1810,340 @@ static inline Obj CacheOper (
 
 #endif
 
+/* This function actually searches the cache. Normally it should be called with 
+   n a compile-time constant to allow the optimiser to tidy things up */
 
-Obj DoOperation0Args (
-    Obj                 oper )
-{
-    Obj                 res;
-    Obj *               cache;
-    Obj                 method;
-    Int                 i;
-    Obj                prec;
 
-    /* try to find an applicable method in the cache                       */
-    prec = INTOBJ_INT(-1);
 
-    do {
-      /* The next line depends on the implementation of INTOBJS */
-      prec = (Obj)(((Int)prec) +4);
-      method = 0;
+static inline Obj __attribute__((always_inline))
```

Comment:
`__attribute__((always_inline))` is not portable, so a `configure` check for it, defining a suitable macro, is needed.